torchrl/data/tensor_specs.py

```diff
@@ -2582,7 +2582,10 @@ def __init__(self, *args, shape=None, device=None, **kwargs):
                     item = CompositeSpec(item, shape=shape)
                 if item is not None:
                     if self._device is None:
-                        self._device = item.device
+                        try:
+                            self._device = item.device
+                        except RuntimeError:
+                            self._device = item._device
```

Comment:
I'm ok with this but maybe let's keep the error and raise it if the item does not have `_device`? Or are we sure it has it?
```suggestion
                        try:
                            self._device = item.device
                        except RuntimeError as err:
                            if some_pattern in str(err):
                                self._device = item._device
                            else:
                                raise err
```

Reply:
If the error is raised it means that item._device is None, in which case we can still clone with None device

Reply:
we are sure that item._device is None so basically we set that